src/main/cpp/processors/denormals.h

```diff
@@ -0,0 +1,15 @@
+// Macro for killing denormalled numbers
+//
+// Written by Jezar at Dreampoint, June 2000
+// http://www.dreampoint.co.uk
+// Based on IS_DENORMAL macro by Jon Watte
+// This code is public domain
+
+#ifndef _denormals_
+#define _denormals_
+
+#define undenormalise(sample) if(((*(unsigned int*)&sample)&0x7f800000)==0) sample=0.0f
+

```

Comment:
By the looks of it, this macro ensures given value is a valid number within expected range.

Probably worthwhile replacing _unsigned int_ with _UINT32_ to be safe across CPU architectures. Also, this looks to be only worknig for SAMPLE_TYPE of _float_ (precision 1), judging from the _0x7f800000_ value.

This could perhaps be better written as:

```
#if PRECISION == 1
#define undenormalise(sample) ((((*(UINT32 *)&(sample))&0x7f800000)==0)&&((sample)!=0.f))
#else
#define undenormalise(sample) ((((((UINT32 *)&(sample))[1])&0x7fe00000)==0)&&((sample)!=0.))	
#end
```